tests/extension/khr_group_interface/group_interface.cpp

```diff
@@ -0,0 +1,399 @@
+  passed &= (group.get_group_id() == work_group.id());
+
+  // linear_id
+  static_assert(std::is_same_v<
+                decltype(work_group.linear_id()),
+                typename sycl::khr::work_group<Dimensions>::linear_id_type>);
+  passed &= (group.get_group_linear_id() == work_group.linear_id());
+
+  // range
+  static_assert(
+      std::is_same_v<decltype(work_group.range()),
+                     typename sycl::khr::work_group<Dimensions>::range_type>);
+  passed &= (group.get_group_range() == work_group.range());
+
+#if __cplusplus >= 202302L
+  // extents
+  static_assert(
+      std::is_same_v<decltype(work_group.extents()),
+                     typename sycl::khr::work_group<Dimensions>::extents_type>);
+  {
+    const sycl::range<Dimensions> localRange = group.get_local_range();
+    if constexpr (Dimensions == 1)
+      passed &= (work_group.extents() ==
+                 std::dextents<std::size_t, Dimensions>(localRange[0]));
+    else if constexpr (Dimensions == 2)
+      passed &= (work_group.extents() == std::dextents<std::size_t, Dimensions>(
+                                             localRange[0], localRange[1]));
+    else if constexpr (Dimensions == 3)
+      passed &= (work_group.extents() ==
+                 std::dextents<std::size_t, Dimensions>(
+                     localRange[0], localRange[1], localRange[2]));
+  }
+
+  // extent
+  static_assert(std::is_same_v<decltype(work_group.extent(0)),
+                               typename sycl::khr::work_group<
+                                   Dimensions>::extents_type::index_type>);
+  for (int i = 0; i < Dimensions; i++)
+    passed &= (work_group.extent(i) == work_group.extents().extent(i));
+
+  // rank
+  static_assert(
+      std::is_same_v<
+          decltype(sycl::khr::work_group<Dimensions>::rank()),
+          typename sycl::khr::work_group<Dimensions>::extents_type::rank_type>);
+  static_assert(decltype(work_group)::rank() ==
+                decltype(work_group.extents())::rank());
+
+  // rank_dynamic
+  static_assert(
+      std::is_same_v<
+          decltype(sycl::khr::work_group<Dimensions>::rank_dynamic()),
+          typename sycl::khr::work_group<Dimensions>::extents_type::rank_type>);
+  static_assert(decltype(work_group)::rank_dynamic() ==
+                decltype(work_group.extents())::rank_dynamic());
+
+  // static_extent
+  static_assert(std::is_same_v<
+                decltype(sycl::khr::work_group<Dimensions>::static_extent(0)),
+                std::size_t>);
+  static_assert(decltype(work_group)::static_extent(0) ==
+                decltype(work_group.extents())::static_extent(0));
+  if (Dimensions >= 2)
+    static_assert(decltype(work_group)::static_extent(1) ==
+                  decltype(work_group.extents())::static_extent(1));
+  if (Dimensions == 3)
+    static_assert(decltype(work_group)::static_extent(2) ==
+                  decltype(work_group.extents())::static_extent(2));
```

Comment:
How can this work with an `if` instead of an `if constexpr`? :thinking:

Reply:
Another solution could be to use a helper function to do the `static_assert` in it, like
```c++
template <int I, typename WG>
void check_static_assert(const WG& work_group) {
 static_assert(WG::static_extent(I) ==
                  decltype(work_group.extents())::static_extent(I));
}
[...]
  if constexpr (Dimensions >= 2)
    check_static_assert<1>(work_group);
  if (Dimensions == 3)
    check_static_assert<2>(work_group);
```